### PR TITLE
initial endpoint to deploy Version to new Environment

### PR DESCRIPTION
**Pull request by jlaswell**

### CODE REVIEW

**Comment by kyleferguson** [comment]
@jlaswell I've pushed up my integration into your branch. I'm sorry I wasn't able to get this working all the way through but I think it's really close. The current error I'm getting is `Error deploying version: 'Deployment.extensions "payments-production" is invalid: spec.template.spec.containers: Required value`.

You should be able to see the flow of things pretty easily. The version details page has the form to deploy to a new environment. The main change is in your original implementation a new version was created, when really it's the same version it should just be deployed to a new environment. Hopefully it will only take a couple quick fixes to get this working.

If we can get this branch in a workable state then we're sprint complete. The data refactor that @larkmullins is working on is a bonus. If you all run into any major blockers with this just ping me today.

**Review by kyleferguson — APPROVED** [review]
Proceed as necessary :)

File: resources/environment.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func UpdateEnvironmentService(db *gorm.DB, client kube.APIInterface, environment
 // GetEnvironment will get an existing Environment by id
 func GetEnvironment(db *gorm.DB, id string) (*Environment, error) {
 	environment := new(Environment)
-	if err := db.Preload("Application").Preload("Versions").Where("uuid = ?", id).First(&environment).Error; err != nil {
+	if err := db.Preload("Application.Environments").Preload("Versions").Where("uuid = ?", id).First(&environment).Error; err != nil {
```
Comment (kyleferguson) [inline resources/environment.go:149]: This is in order to show the available environments to deploy to on the version details page. Not ideal but it works.

**Comment by jlaswell** [comment]
So long as CI passes, this is feature complete. 🎉

**Review by larkmullins — APPROVED** [review]

**Pull request merged**